cranelift/codegen/src/isa/s390x/inst/args.rs

```diff
@@ -88,6 +91,21 @@ impl MemArg {
         MemArg::RegOffset { reg, off, flags }
     }
 
+    /// Add an offset to a virtual addressing mode.
+    pub fn offset(base: &MemArg, offset: i64) -> MemArg {
+        match base {
+            &MemArg::RegOffset { reg, off, flags } => MemArg::RegOffset {
+                reg,
+                off: off + offset,
+                flags,
+            },
+            &MemArg::InitialSPOffset { off } => MemArg::InitialSPOffset { off: off + offset },
+            &MemArg::NominalSPOffset { off } => MemArg::NominalSPOffset { off: off + offset },
+            &MemArg::SlotOffset { off } => MemArg::SlotOffset { off: off + offset },
+            _ => unreachable!(),
```

Comment:
What cases is this matching? I think it might be better to explicitly list them, rather than using `_`, so that if more `MemArg` variants are added in the future we don't forget to consider/update this case.